File: 0724-find-pivot-index/0724-find-pivot-index.cpp
class Solution {
public:
    int pivotIndex(vector<int>& nums) {
        int n = nums.size();
        int total = accumulate(nums.begin(), nums.end(),0), leftSum=0;
        for(int i=0;i<n;i++){
            if(2*leftSum==total-nums[i])   return i;
            leftSum+=nums[i];
        }
        return -1;
    }
};

File: 0724-find-pivot-index/0724-find-pivot-index.py
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        n = len(nums)
        leftSum = 0
        rightSum = sum(nums[1:])
        for i in range(n):
            if leftSum == rightSum:
                return i
            leftSum += nums[i]
            if i+1 < n:
                rightSum -= nums[i+1]
        return -1